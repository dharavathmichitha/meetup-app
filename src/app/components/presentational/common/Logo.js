import React from 'react';
import PropTypes from 'prop-types';

const Logo = ({ smallText, largeText }) => {
  return (
    <div className="topheader__logo">
      <span className="topheader__logo--small">{smallText}</span>
      <span className="topheader__logo--large">{largeText}</span>
    </div>
  );
};

Logo.defaultProps = {
  smallText: 'my',
  largeText: 'Logo',
};

Logo.propTypes = {
  smallText: PropTypes.string,
  largeText: PropTypes.string,
};

export default Logo;
