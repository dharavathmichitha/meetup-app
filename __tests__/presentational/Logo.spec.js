import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../../src/app/components/presentational/common/Logo';

beforeEach(() => {
  jest.resetModules();
});

describe('Logo Component', () => {
  it('should match snapshot including default props', () => {
    const props = {
      smallText: 'my',
      largeText: 'Logo',
    };
    const component = renderer.create(<Logo {...props} />);

    const componentTree = component.toJSON();

    expect(componentTree).toMatchSnapshot();
  });
});
