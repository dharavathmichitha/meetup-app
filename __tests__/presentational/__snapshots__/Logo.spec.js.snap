// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component should match snapshot including default props 1`] = `
<div
  className="topheader__logo"
>
  <span
    className="topheader__logo--small"
  >
    my
  </span>
  <span
    className="topheader__logo--large"
  >
    Logo
  </span>
</div>
`;
